conda/common/logic.py

```diff
@@ -364,12 +364,12 @@ def _convert(self, x):
         except KeyError:
             raise ValueError("Unregistered SAT variable name: {}".format(name))
 
-    def _eval(self, func, args, polarity, name):
+    def _eval(self, func, args, no_literal_args, polarity, name):
```

Comment:
In this commit we re-add `LinearBound`. That method has, in contrast to the other ones, not only arguments that are literals (or lists thereof or lists of clauses which are just tuples of literals) but coefficients (i.e., costs/weights). Those coefficients are also integers but do not represent SAT variables. Hence we should not run `_convert` on those.
Previously, `LinearBound` would just omit `_convert` altogether. Now we separate the arguments into `args` with literals and `no_literal_args` for other arguments like coefficients and (non-literal) Booleans.